### ISSUE TITLE
Cannot wait on RunningCommand twice

### ISSUE THREAD

**Issue by virtuald** [amoffat/sh#303]
Consider this example:

```
x = sh.app(_bg=True)
try:
    x.wait()
except KeyboardInterrupt:
    x.wait()
```

In this case, the second wait will return immediately, instead of waiting for the child to exit.

The workaround of course is the following, but it would be good if the workaround was not required:

```
x = sh.app(_bg=True)
try:
    x.wait()
except KeyboardInterrupt:
    x._process_completed = False
    x.wait()
```

**Comment by amoffat** [comment]
Thanks for reporting @virtuald

**Comment by amoffat** [comment]
This appears to be fixed on the `release-1.2` branch and will ship when that ships.  Thanks again for the report.
